tpm/tpm.go

```diff
@@ -904,6 +904,61 @@ func ReadEKCert(rw io.ReadWriter, ownAuth digest) ([]byte, error) {
 	return ekbuf, nil
 }
 
+// NVDefineSpace defines space in NVRAM for the indicated index.
+// Definition include the access requirements for writing and reading the area
+func NVDefineSpace(rw io.ReadWriter, index tpmutil.Handle, len uint32, perms []uint32, ownAuth digest) ([]byte, error) {
+	sharedSecretOwn, osaprOwn, err := newOSAPSession(rw, etOwner, khOwner, ownAuth[:])
+	if err != nil {
+		return nil, fmt.Errorf("failed to start new auth session: %v", err)
+	}
+	defer osaprOwn.Close(rw)
+	defer zeroBytes(sharedSecretOwn[:])
+	perm := uint32(0)
+	for _, i := range perms {
+		perm += i
+	}
+	nvdata := nvDataPublic{0x0018, index, pcrInfoShort{}, pcrInfoShort{}, nvAttribute{tpmutil.Tag(0x0017), uint32(perm)}, false, false, false, len}
```

Comment:
Please add information about the hard-coded constants here. Can they be made into named constants in this file?